code/datums/composite_sounds/_composite_sound.dm

```diff
@@ -0,0 +1,90 @@
+/*
+	output_atoms	(list of atoms)			The destination(s) for the sounds
+	mid_sounds		(list or soundfile)		Since this can be either a list or a single soundfile you can have random sounds. May contain further lists but must contain a soundfile at the end.
+	mid_length		(num)					The length to wait between playing mid_sounds
+	start_sound		(soundfile)				Played before starting the mid_sounds loop
+	start_length	(num)					How long to wait before starting the main loop after playing start_sound
+	end_sound		(soundfile)				The sound played after the main loop has concluded
+	chance			(num)					Chance per loop to play a mid_sound
+	volume			(num)					Sound output volume
+	max_loops		(num)					The max amount of loops to run for.
+	direct			(bool)					If true plays directly to provided atoms instead of from them
+*/
+
+/datum/composite_sound
+	var/list/atom/output_atoms
+	var/mid_sounds
+	var/mid_length
+	var/start_sound
+	var/start_length
+	var/end_sound
+	var/chance
+	var/volume = 100
+	var/max_loops
+	var/direct
+
+	var/timerid
+
+/datum/composite_sound/New(list/_output_atoms=list(), start_immediately=FALSE, _direct=FALSE)
+	if(!mid_sounds)
+		WARNING("A looping sound datum was created without sounds to play.")
+		return
+
+	output_atoms = _output_atoms
+	direct = _direct
+
+	if(start_immediately)
+		start()
+
+/datum/composite_sound/Destroy()
+	stop()
+	output_atoms = null
+	return ..()
+
+/datum/composite_sound/proc/start(atom/add_thing)
+	if(add_thing)
+		output_atoms |= add_thing
```

Comment:
LAZYDISTINCTADD possibly, this will runtime if the sound was created with no output atoms.